newrelic-agent/src/test/java/com/newrelic/agent/RPMServiceTest.java

```diff
@@ -586,7 +585,54 @@ public void transactionNamingScheme() throws Exception {
         assertEquals(TransactionNamingScheme.LEGACY, svc.getTransactionNamingScheme());
     }
 
-    // TODO sendLogEvents & sendLogEventsWithPut tests?
+    @Test(timeout = 30000)
+    public void testSendLogEvents() throws Exception {
+        Map<String, Object> map = createStagingMap(true, false);
+        map.put("app_name", "Test");
+        AgentConfig config = AgentConfigImpl.createAgentConfig(map);
+        createServiceManager(config, map);
+        doSendLogEvent();
+    }
+
+    @Test(timeout = 30000)
+    public void testSendLogEventsWithPut() throws Exception {
+        Map<String, Object> map = createStagingMap(true, false, true);
+        map.put("app_name", "Test");
+        AgentConfig config = AgentConfigImpl.createAgentConfig(map);
+        createServiceManager(config, map);
+        doSendLogEvent();
+    }
+
+    private void doSendLogEvent() throws Exception {
+        MockDataSenderFactory dataSenderFactory = new MockDataSenderFactory();
+        DataSenderFactory.setDataSenderFactory(dataSenderFactory);
+        List<String> appNames = singletonList("Send Log Events Test App");
+        RPMService svc = new RPMService(appNames, null, null, Collections.<AgentConnectionEstablishedListener>emptyList());
+        ProfilerParameters parameters = new ProfilerParameters(0L, 0L, 0L, false, false, Agent.isDebugEnabled(), null,
+                null);
+        Profile profile = new Profile(parameters);
+        List<IProfile> profiles = new ArrayList<>();
+        profiles.add(profile);
+        ProfileSampler sampler = new ProfileSampler();
+        sampler.sampleStackTraces(profiles);
+
+        profile.start();
+        profile.end();
```

Comment:
This whole section can be removed as we're not testing the thread profiler here.

```
        ProfilerParameters parameters = new ProfilerParameters(0L, 0L, 0L, false, false, Agent.isDebugEnabled(), null,
                null);
        Profile profile = new Profile(parameters);
        List<IProfile> profiles = new ArrayList<>();
        profiles.add(profile);
        ProfileSampler sampler = new ProfileSampler();
        sampler.sampleStackTraces(profiles);

        profile.start();
        profile.end();
```

Reply:
Cool, honestly didn't know all that was going on and if it might be needed for something under the covers.  I'll fix this ASAP.

Reply:
Yea there's a lot to make sense of. This was just logic to generate thread profile data for a different agent endpoint in the same way that you're generating log event data for this endpoint.